src/geo/projection/globe.js

```diff
@@ -218,7 +218,7 @@ export function globeECEFUnitsToPixelScale(worldSize: number) {
     return wsRadius / localRadius;
 }
 
-export function calculateGlobeMatrix(tr: Transform, worldSize: number, offset?: [number, number]): mat4 {
+export function calculateGlobeMatrix(tr: Transform, worldSize: number, offset?: [number, number]): Float32Array {
```

Comment:
I believe reducing precision of some of the intermediate matrices from 64 bits to 32 bits will become problematic. Globe matrix is pretty much equivalent of tile matrices (`createTileMatrix` for other projections) meaning that it requires high precision for representing pixel coordinates at all zoom levels. Any precision issues might be masked though by the low zoom threshold value we have for the globe view but it will eventually bite us back.

Reply:
Yeah, that's the moment I want to check up on — which of these matrices are intermediary and need high precision? I tried to keep them `Float64Array` but had a hard time reconciling typing errors when some of these matrices eventually get passed to WebGL methods that expect `Float32Array`.

Reply:
@mpulkki-mapbox I reverted globe matrix to double precision — any other matrices I need to do this for?

Reply:
The rule of thumb would be that any matrix that is multiplied with `transform.projMatrix` should be kept 64bit. I'll go through the list quickly.

Reply:
@mourner: it's actually quite difficult to tell whether current type annotations for matrices are correct or not. For example the function signature `calculateProjMatrix(unwrappedTileID: UnwrappedTileID, aligned: boolean = false): Float32Array` in transform.js actually returns 64bit matrices as all possible code paths inside the function returns `Float64Arrays`. Similar patterns exists in other parts too. Or am I overlooking something perhaps?

Reply:
@mpulkki-mapbox how so? It's explicitly returning `Float32Array` at the moment: 

https://github.com/mapbox/mapbox-gl-js/blob/74559af9376932d45f3e475ad0b499438147a5a4/src/geo/transform.js#L1444-L1445

The new type mismatches in the new branch are caught pretty well — certainly a huge improvement over silently passing mistypes in `main`.

Reply:
oh you're correct! The conversion is done at the end of the function call :). It's perhaps the `calculatePosMatrix` I was looking at. That function will forward the matrix creation to a specific TileTransform class and both FlatTileTransform and GlobeTileTransform will return 64 bit matrices. Am I looking things correctly now? :)

Reply:
@mpulkki-mapbox I reverted `createTileMatrix` to `Float64Array`, so now it should at least be consistent. I guess it can't fully figure out the indirection in this instance to see that it doesn't match the type defined in `projection/index.js`, but upgrading Flow to the latest (2.5 years newer) version with much more sound typing architecture (which is my end goal) might help.